### PR TITLE
Whios unknown `registrar` key

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/44957

## Description
Fixed an issue on URL's with EU suffix

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Whois/Integrations/Whois/test_data/EU domains.text [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        Name: IONOS SE
+        Website: https://ionos.com
+
+Name servers:
+        ns1060.ui-dns.biz
+        ns1060.ui-dns.com
+        ns1060.ui-dns.org
+        ns1060.ui-dns.de
+
+Please visit www.eurid.eu for more info.
```
Comment (reutshal) [inline Packs/Whois/Integrations/Whois/test_data/EU domains.text:67]: Where did we get this text from?

File: Packs/Whois/ReleaseNotes/1_2_8.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Whois
+- Fixed an issue where the ***Whois*** command failed in domains with eu suffix
```
Comment (reutshal) [inline Packs/Whois/ReleaseNotes/1_2_8.md:4]:
```suggestion
- Fixed an issue where the ***Whois*** command mishandled domains with 'eu' suffix.
```

**Review by reutshal — CHANGES_REQUESTED** [review]

File: Packs/Whois/Integrations/Whois/Whois.py [hunk 3]
```diff
@@ -7742,7 +7742,8 @@ def parse_raw_whois(raw_data, normalized=None, never_query_handles=True, handle_
         # SIDN isn't very standard either. And EURid uses a similar format.
         match = re.search("Registrar:\n\s+(?:Name:\s*)?(\S.*)", segment)
         if match is not None:
-            data["registrar"].insert(0, match.group(1).strip())
+            # Set default value -> https://docs.python.org/3/library/stdtypes.html#dict.setdefault
+            data.setdefault("registrar", []).insert(0, match.group(1).strip())
```
[the remark this answers is not in the document]
Comment (MosheEichler) [reply Packs/Whois/Integrations/Whois/Whois.py:7746]: This just gives us a list in default, but we want to get/set (get if exist and set if not) a list in `registrar` key

File: Packs/Whois/Integrations/Whois/test_data/EU domains.text (hunk 1 above)
Comment (MosheEichler) [reply Packs/Whois/Integrations/Whois/test_data/EU domains.text:67]: It's returned from `whois` with `sv-online.eu` domain
https://who.is/whois/sv-online.eu

File: Packs/Whois/Integrations/Whois/Whois.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (reutshal) [reply Packs/Whois/Integrations/Whois/Whois.py:7746]: Why are we setting this? are we getting the 'registrar' key in the rest of the integration?

[the remark this answers is not in the document]
Comment (MosheEichler) [reply Packs/Whois/Integrations/Whois/Whois.py:7746]: Yes it's a part of the context data
Screen Shot 2021-12-28 at 14 55 58

[the remark this answers is not in the document]
Comment (reutshal) [reply Packs/Whois/Integrations/Whois/Whois.py:7746]: OK, I just didn't get something - if we are setting an empty list here, how come there is a value at the end? (Looking at the unit test)

**Review by reutshal — APPROVED** [review]

File: Packs/Whois/ReleaseNotes/1_2_8.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Whois
+- Fixed an issue where the ***Whois*** command mishandled domains with 'eu' suffix.
```
Comment (amshamah419) [inline Packs/Whois/ReleaseNotes/1_2_8.md:4]:
```suggestion
- Fixed an issue where the ***whois*** command would mishandle domains with an 'eu' suffix.
```

File: Packs/Whois/Integrations/Whois/test_data/EU domains.text [hunk 5]
```diff
@@ -0,0 +1,67 @@
+% The WHOIS service offered by EURid and the access to the records
```
Comment (reutshal) [inline Packs/Whois/Integrations/Whois/test_data/EU domains.text:1]: Do we need the whole file for testing?

Comment (MosheEichler) [reply Packs/Whois/Integrations/Whois/test_data/EU domains.text:1]: I think so,
The `parse_raw_whois` function gets an argument `raw_data` which is the content of the file.

**Review by reutshal — APPROVED** [review]

**Pull request merged**